Assets/MixedRealityToolkit-Tests/Core/TestFixture_01_MixedRealityToolkitTests.cs

```diff
@@ -0,0 +1,543 @@
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestDataProvider1), new TestExtensionService1("Test16-1.2", 10));
+
+            // Add test 2 services
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestDataProvider2), new TestExtensionService2("Test16-2.1", 10));
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestDataProvider2), new TestExtensionService2("Test16-2.2", 10));
+
+            // Retrieve all extension services.
+            var allExtensionServices = MixedRealityToolkit.Instance.GetActiveServices(typeof(IMixedRealityExtensionService));
+
+            // Tests
+            Assert.AreEqual(4, allExtensionServices.Count);
+        }
+
+        #endregion IMixedRealityDataprovider Tests
+
+        #region IMixedRealityExtensionService Tests
+
+        [Test]
+        public void Test_05_01_RegisterMixedRealityExtensionService()
+        {
+            TestUtilities.InitializeMixedRealityToolkitScene();
+
+            // Register ITestExtensionService1
+            MixedRealityToolkit.Instance.RegisterService(typeof(ITestExtensionService1), new TestExtensionService1("Test ExtensionService 1", 10));
+
+            // Retrieve ITestExtensionService1
+            var extensionService1 = MixedRealityToolkit.Instance.GetService(typeof(ITestExtensionService1));
+
+            // Tests
+            Assert.IsEmpty(MixedRealityToolkit.ActiveSystems);
+            Assert.AreEqual(1, MixedRealityToolkit.RegisteredMixedRealityServices.Count);
+
+            // Tests
+            Assert.IsNotNull(extensionService1);
+        }
+
+        [Test]
+        public void Test_05_02_01_UnregisterMixedRealityExtensionServiceByType()
+        {
+            // Redo register test service for a valid instance.
+            Test_05_01_RegisterMixedRealityExtensionService();
+
+            // Retrieve service
+            var extensionService1 = MixedRealityToolkit.Instance.GetService(typeof(ITestExtensionService1));
+
+            // Validate
+            Assert.IsNotNull(extensionService1);
+
+            var success = MixedRealityToolkit.Instance.UnregisterService(typeof(ITestExtensionService1));
+
+            // Validate non-existent service
+            var isServiceRegistered = MixedRealityToolkit.Instance.IsServiceRegistered<ITestExtensionService1>();
+
+            // Tests
+            Assert.IsTrue(success);
+            Assert.IsFalse(isServiceRegistered);
+            Assert.IsEmpty(MixedRealityToolkit.ActiveSystems);
+            Assert.IsEmpty(MixedRealityToolkit.RegisteredMixedRealityServices);
+            LogAssert.Expect(LogType.Error, $"Unable to find {typeof(ITestExtensionService1).Name} service.");
+        }
+
+        [Test]
+        public void Test_05_02_02_UnregisterMixedRealityExtensionServiceByTypeAndName()
+        {
+            // Redo register test service for a valid instance.
+            Test_05_01_RegisterMixedRealityExtensionService();
```

Comment:
Rather than re-running a test (which may fail) please consider a helper function that sets the desired initial conditions?